src/common/time/parser/datetime_parser.yy

```diff
@@ -0,0 +1,259 @@
+%language "C++"
+%skeleton "lalr1.cc"
+%no-lines
+%locations
+%define api.namespace { nebula::time }
+%define parser_class_name { DatetimeParser }
+%lex-param { nebula::time::DatetimeScanner& scanner }
+%parse-param { nebula::time::DatetimeScanner& scanner }
+%parse-param { std::string &errmsg }
+%parse-param { nebula::DateTime** output }
+%parse-param { nebula::time::Type outputType }
+
+%code requires {
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <cstddef>
+#include "common/datatypes/Date.h"
+#include "common/time/TimeConversion.h"
+#include "common/time/TimezoneInfo.h"
+#include "common/time/TimeUtils.h"
+
+namespace nebula {
+namespace time {
+class DatetimeScanner;
+
+enum class Type {
+  kDate,
+  kTime,
+  kDateTime,
+};
+}
+}
+
+}
+
+%code {
+    #include "common/time/parser/DatetimeScanner.h"
+    static int yylex(nebula::time::DatetimeParser::semantic_type* yylval,
+                     nebula::time::DatetimeParser::location_type *yylloc,
+                     nebula::time::DatetimeScanner& scanner);
+}
+
+%union {
+    int64_t                                 intVal;
+    double                                  doubleVal;
+    nebula::DateTime                       *dtVal;
+    nebula::Date                           *dVal;
+    nebula::Time                           *tVal;
+    std::string                            *strVal;
+}
+
+/* destructors */
+%destructor {} <intVal> <doubleVal>
+%destructor {} <dtVal>  // for output
+%destructor { delete $$; } <*>
+
+/* keyword */
+%token KW_TIME_ID
+
+/* symbols */
+%token TIME_DELIMITER SPACE POSITIVE NEGATIVE
+
+/* token type specification */
+%token <intVal> INTEGER
+%token <strVal> TIME_ZONE_NAME
+%token <doubleVal> DOUBLE
+
+%type <dtVal> datetime
+%type <dVal> date
+%type <tVal> time
+%type <intVal> opt_time_zone time_zone_offset opt_time_zone_offset opt_time_zone_name
+
+%define api.prefix {datetime}
+
+%start datetime
+
+%%
+
+datetime
+  : date date_time_delimiter time opt_time_zone {
+    if (outputType != nebula::time::Type::kDateTime) {
+      delete $1;
+      delete $3;
+      throw DatetimeParser::syntax_error(@1, "Mismatched date time type.");
+    }
+    $$ = new DateTime(TimeConversion::dateTimeShift(DateTime(*$1, *$3), -$4));
+    *output = $$;
+    delete $1;
+    delete $3;
+  }
+  | date {
+    if (outputType != nebula::time::Type::kDate) {
+      delete $1;
+      throw DatetimeParser::syntax_error(@1, "Mismatched date time type.");
+    }
+    $$ = new DateTime(*$1);
+    *output = $$;
+    delete $1;
+  }
```

Comment:
```
neo4j@neo4j> RETURN datetime("2021-11-13");
+------------------------+
| datetime("2021-11-13") |
+------------------------+
| 2021-11-13T00:00Z      |
+------------------------+
```

Reply:
It seems that this behavior is inconsistent with `neo4j`.

Reply:
It's conflict with date rule.